tasks/components.py

```diff
@@ -179,6 +184,29 @@ def get_components_and_bundles():
     return sorted(sorted_bundles), sorted(components_without_bundle), ok
 
 
+class ComponentRoot(object):
+    def __init__(self, file, dir, version):
+        self.file = file
+        self.dir = dir
+        self.version = version
+        self.content = list(self.file.open())
+
+
+def locate_root(dir):
+    """
+    Locate the root of the component, if this directory contains a component
+    """
+    # v2 component: this folder is a component root if it contains 'def/component.go'
+    component_file = dir / 'def/component.go'
```

Comment:
Did you check that using `def` folder allows IDEs to add import automatically on a real components?

Reply:
It works for me using `gofumpt` instead of `gopls` configured in VScode. I haven't dug deeply enough to figure out exactly why it works.